### DISCUSSION TITLE
How to properly export plain objects with methods (like Math)?

### DISCUSSION THREAD

**Discussion by krutoo** [jsr-io/jsr#633] [Q&A]
Hi, i noticed that JSR now throws error for public objects of package like this:

```ts
export const MathExtras = {
  randomInt (min: number, max: number): number {
    // some implementation here
  },
} as const;
```

Error says that MathExtras should have explicit type.

As I understand, I have to define type or interface and bind it to `MathExtras` contant, but I dont want to declare redundant types/interfaces.

How is it proposed to declare such objects for JSR?

**Comment by pukingrainbows** [comment]
You are not explicitly typing the return `MathExtras` object and `as const` is not enough information for jsr.io to inferred what you are trying to return, jsr.io expected you to tell what is the return type, since it will not load run `ts` in order to emit declaration, this declaration are statically analyze. [about slow types](https://jsr.io/docs/about-slow-types#what-are-slow-types)


something like this should work?

```typescript
interface IMathExtras {
  randomInt: (min: number, max: number) => number
}

export const MathExtras: IMathExtras = {
  randomInt(min, max) {
     return 1;
  }
}
```

**Reply by krutoo** [reply]
@pukingrainbows yes, this is exactly what I mean by "redundant types/interfaces"

In fact, randomInt method has full signature and MathExtras too

In real cases I have maybe dozens of methods in dozens of such objects

I also need to export redudant interface by some specific cases because of TypeScript

**Comment by mybearworld** [comment]
You seem to really just be simulating a module here - maybe you could declare it as such?

```ts
// mathExtras.ts

export const randomInt = (min: number, max: number) => {
  // do something
}
```
